### PR TITLE
Improve behaviour of "check for updates" button

### PR DESCRIPTION
**Pull request by peppy**
- Now lets the user know if they are already on the latest version.
- Squirrel updater will re-post the "click to restart/update" notification, in case the user missed / dismissed it.

I was looking into this because the "check for updates" button didn't seem to work on windows. I get the feeling it was in the above state, so want to get this in and see if it fixes the behaviour in future builds, or if there's something else at play. In local testing I couldn't notice any other fail scenarios.

### CODE REVIEW

**Comment by enoslayd** [comment]
i tried this pr and it seems to update even though it is running a non-release version (which before (master) doesnt)

**Comment by peppy** [comment]
Good catch. I think I left some debug code in.

**Review by bdach — COMMENTED** [review]
tested the Linux path using `osu-deploy`, seems to be behaving as expected :+1:

File: osu.Game/Updater/UpdateManager.cs [hunk 1]
```diff
@@ -57,25 +57,28 @@ protected override void LoadComplete()
...
+
+            return hasUpdates;
         }
 
-        protected virtual Task PerformUpdateCheck() => Task.CompletedTask;
+        /// <summary>
+        /// Performs an asynchronous check for application updates.
+        /// </summary>
+        /// <returns>Whether any update is waiting. May return true if an error occured (there is potentially an update available).</returns>
+        protected virtual Task<bool> PerformUpdateCheck() => Task.FromResult(false);
```
[the remark this answers is not in the document]
Comment (peppy) [reply osu.Game/Updater/UpdateManager.cs:81]: if it was more exposed code i'd have opted for a ternary/enum of some kind. but i still don't know how final the update system is so not putting too much thought into it.

**Comment by enoslayd** [comment]
> i tried this pr and it seems to update even though it is running a non-release version (which before (master) doesnt)
> image
> image

it still happens
image

File: osu.Game/Updater/UpdateManager.cs [hunk 2]
```diff
@@ -57,25 +57,28 @@ protected override void LoadComplete()
 
         private readonly object updateTaskLock = new object();
 
-        private Task updateCheckTask;
+        private Task<bool> updateCheckTask;
 
-        public async Task CheckForUpdateAsync()
+        public async Task<bool> CheckForUpdateAsync()
         {
-            if (!CanCheckForUpdate)
```
Comment (bdach) [inline osu.Game/Updater/UpdateManager.cs:64]: this should probably stay to resolve the above

**Review by bdach — APPROVED** [review]

**Pull request merged**